### ISSUE TITLE
Form Status Changes observable, should also emit the Value used for validation.

### ISSUE THREAD

**Issue by erkinyldz** [angular/angular#45334]
### Which @angular/* package(s) are relevant/releated to the feature request?

forms

### Description

We have a requirement to make a call to API every time the value of form is changed, we need to make an API call, if it is valid. Form is a complex form with multiple child groups and child arrays.

Since valueChanges is fired before statusChanges and since statusChanges can be async; it is not possible to do this in a reliable way.

### Proposed solution

statusChanges event should also emit the value used for calculating the status.

### Alternatives considered

To solve this issue, I am using the following observable. This is not ideal, however I could not think of a better way.

```
const filterSetValidityChanged$ = this.searchForm.get('filterSet').statusChanges.pipe();
const filterSetChanged$ = this.searchForm.get('filterSet').valueChanges.pipe();

const filterSetChangedAndIsValid$ = filterSetValidityChanged$.pipe(
  filter(a => a === 'VALID'),
  withLatestFrom(filterSetChanged$),
  map(([, value]) => value));
```

**Comment by dylhunn** [comment]
This might be solvable in the context of #10887, if we had a unified event payload that contained both the `value` and `status` for all state changes.

**Comment by Totati** [comment]
Why do you need `withLatestFrom`? It seems to me that you could write a little helper function that does the job.
[Like](https://stackblitz.com/edit/angular-ivy-valid-control-values?file=src%2Fapp%2Fapp.component.ts):
``` javascript
function validValues(control: AbstractControl) {
  return control.statusChanges.pipe(
    filter((a) => a === 'VALID'),
    map(() => control.value),
  );
}
```

**Comment by dylhunn** [comment]
> Why do you need `withLatestFrom`? It seems to me that you could write a little helper function that does the job. [Like](https://stackblitz.com/edit/angular-ivy-valid-control-values?file=src%2Fapp%2Fapp.component.ts):
> 
> ```js
> function validValues(control: AbstractControl) {
>   return control.statusChanges.pipe(
>     filter((a) => a === 'VALID'),
>     map(() => control.value),
>   );
> }
> ```

I think the issue is that the value might have changed in the meantime, since the validation is async and could take a while.

**Comment by Totati** [comment]
Does not seem to effect that. I've updated my stackblitz.

**Comment by erkinyldz** [comment]
I have actually had to spent more time for the issue I had and dived into source code. Status is always and always fired right after values are fired in the current implementation. However that implementation might easily change and the current events does not guarantee a way to get values that has been used for validation. I still think this request is a good idea, and gives consumers a way to get which values are used for validation.
@Totati what you give suggested, or what I did will work for current implementation, however if underlying implementations change, this will not work. That is the main problem I am pointing at.
